### ISSUE TITLE
Add Servo

### ISSUE THREAD

**Issue by martinohanlon** [RaspberryPiFoundation/picozero#75]
**Describe the solution you'd like**

Add `Servo` as a supported device.

**Additional context**

Servo could inherit from `PWMOutputDevice`. To cater for a wide range of devices it accept should min and max pulse width and a frame width in the constructor.

I have created a quick prototype.

```python
from picozero import PWMOutputDevice
from time import sleep

def clamp(n, low, high): return max(low, min(n, high))

class Servo(PWMOutputDevice):
    def __init__(self, pin, initial_value=None, min_pulse_width=1/1000, max_pulse_width=2/1000, frame_width=20/1000, duty_factor=65535):
               
        self._min_duty = int((min_pulse_width / frame_width) * duty_factor)
        self._max_duty = int((max_pulse_width / frame_width) * duty_factor)
        
        super().__init__(pin, freq=int(1 / frame_width), duty_factor=duty_factor, initial_value=initial_value)
        
    def _state_to_value(self, state):
        return None if state == 0 else clamp((state - self._min_duty) / (self._max_duty - self._min_duty), 0, 1)
        
    def _value_to_state(self, value):
        return 0 if value is None else int(self._min_duty + ((self._max_duty - self._min_duty) * value))
    
    def min(self):
        self.value = 0
    
    def mid(self):
        self.value = 0.5
        
    def max(self):
        self.value = 1

s = Servo(28, min_pulse_width=0.7/1000, max_pulse_width=2.1/1000)

s.value = 0
sleep(1)

s.value = 0.5
sleep(1)

s.value = 1
sleep(1)

s.value = None
```

This uses a value between 0 & 1 to set / read a value between min and max positions. Setting value to `None` will turn the servo off (and allow it to be turned). `min`, `mid` and `max` functions allow the servo to be set to the appropriate position.

**Questions**

Should value be a range from 0 to 1? Or (as gpiozero implements it) -1 to 1? I favour 0-1 mainly because it seems to fit with other devices but I have little experience of using servos and I am not sure what the benefits of using -1 to 1 might be.

Angles could also be included, but it would make the constructor a lot more complicated. gpiozero does this via a seperate class [AngularServo](https://gpiozero.readthedocs.io/en/stable/api_output.html#angularservo)

**Comment by martinohanlon** [comment]
I have implemented Servo. The code is currently in the `servo` branch.

**Comment by martinohanlon** [comment]
Servo has been included in the v0.4.0
